tests/integ/scala/test_function_suite.py

```diff
@@ -593,21 +593,64 @@ def test_datediff_negative(session):
         TestData.timestamp1(session).select(dateadd(7, lit(1), col("a")))
 
 
-def test_dateadd(session):
+@pytest.mark.localtest
+@pytest.mark.parametrize(
+    "part,expected",
+    [
+        ("yyy", [Row(date(2021, 8, 1)), Row(date(2011, 12, 1))]),
+        ("qtr", [Row(date(2020, 11, 1)), Row(date(2011, 3, 1))]),
+        ("mon", [Row(date(2020, 9, 1)), Row(date(2011, 1, 1))]),
+        ("wk", [Row(date(2020, 8, 8)), Row(date(2010, 12, 8))]),
+        ("day", [Row(date(2020, 8, 2)), Row(date(2010, 12, 2))]),
+        ("hrs", [Row(datetime(2020, 8, 1, 1, 0)), Row(datetime(2010, 12, 1, 1, 0))]),
+        ("min", [Row(datetime(2020, 8, 1, 0, 1)), Row(datetime(2010, 12, 1, 0, 1))]),
+        (
+            "sec",
+            [Row(datetime(2020, 8, 1, 0, 0, 1)), Row(datetime(2010, 12, 1, 0, 0, 1))],
+        ),
+        (
+            "msec",
+            [
+                Row(datetime(2020, 8, 1, 0, 0, 0, 1000)),
+                Row(datetime(2010, 12, 1, 0, 0, 0, 1000)),
+            ],
+        ),
+        (
+            "usec",
+            [
+                Row(datetime(2020, 8, 1, 0, 0, 0, 1)),
+                Row(datetime(2010, 12, 1, 0, 0, 0, 1)),
+            ],
+        ),
+        (
+            "nsec",
+            [
+                Row(datetime(2020, 8, 1, 0, 0, 0, 10)),
+                Row(datetime(2010, 12, 1, 0, 0, 0, 10)),
+            ],
+        ),
+    ],
+)
+def test_dateadd(part, expected, session):
+    val = 10000 if part == "nsec" else 1
+
+    df = TestData.date1(session)
+
     Utils.check_answer(
-        [Row(date(2021, 8, 1)), Row(date(2011, 12, 1))],
-        TestData.date1(session).select(dateadd("year", lit(1), col("a"))),
+        df.select(dateadd(part, lit(val), col("a"))),
+        expected,
         sort=False,
     )
 
     # Same as above, but pass str instead of Column
     Utils.check_answer(
-        [Row(date(2021, 8, 1)), Row(date(2011, 12, 1))],
-        TestData.date1(session).select(dateadd("year", lit(1), "a")),
+        df.select(dateadd(part, lit(val), "a")),
```

Comment:
NIT: should we have some tests with timezoned dates?

Reply:
I think we should. That will require the completion of SNOW-984699 which I'm currently working on. It's probably worthwhile to defer this PR until that change makes it in.

Reply:
Tests have been added.